BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/service/PendingSubmissionListFilter.java

```diff
@@ -0,0 +1,82 @@
+package uk.ac.ebi.biostd.webapp.application.rest.service;
+
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionListFiltersDto;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionListItemDto;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class PendingSubmissionListFilter {
+
+    public static Predicate<? super PendingSubmissionListItemDto> asPredicate(PendingSubmissionListFiltersDto filters) {
+        List<Predicate<? super PendingSubmissionListItemDto>> predicates = new ArrayList<>();
+
+        Optional.ofNullable(filters.getAccNo())
+                .map(String::trim)
+                .filter(v -> !v.isEmpty())
+                .ifPresent(v -> predicates.add(accNoFilter(v)));
+
+        Optional.ofNullable(filters.getKeywords())
+                .map(String::trim)
+                .filter(v -> !v.isEmpty())
+                .ifPresent(v -> predicates.add(keywordsFilter(v)));
+
+        Optional.ofNullable(filters.getRTimeFrom()).ifPresent(v -> predicates.add(rTimeFromFilter(v)));
+        Optional.ofNullable(filters.getRTimeTo()).ifPresent(v -> predicates.add(rTimeToFilter(v)));
+
+        return item -> predicates.stream().allMatch(p -> p.test(item));
+    }
+
+    private static Predicate<? super PendingSubmissionListItemDto> rTimeFromFilter(final Long dateInSeconds) {
+        return (Predicate<PendingSubmissionListItemDto>) dto -> {
+            Long rtime = dto.getRtime();
+            return rtime != null && rtime.compareTo(dateInSeconds) >= 0;
+        };
+    }
+
+    private static Predicate<? super PendingSubmissionListItemDto> rTimeToFilter(final Long dateInSeconds) {
```

Comment:
can be formatted as (same for other cases):

```
private static Predicate<? super PendingSubmissionListItemDto> rTimeToFilter(Long dateInSeconds) {
        return dto -> {
            Long rtime = dto.getRtime();
            return rtime != null && rtime.compareTo(dateInSeconds) < 0;
        };
    }

```

Reply:
Agree. It left from idea code generation.. Done.